RAG_supporters/dataset/label_calculator.py

```diff
@@ -0,0 +1,383 @@
+"""Label calculation system for domain assessment dataset.
+
+Calculates 3 types of labels:
+1. Source/question labels: Based on CSV cluster_probabilities or centroid distances
+2. Steering embedding labels: Based on centroid distances
+3. Combined labels: Average of source and steering labels (for masking augmentation)
+"""
+
+import logging
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Dict, List, Optional, Tuple, Union
+
+import numpy as np
+from scipy.spatial.distance import cdist
+
+from RAG_supporters.clustering.clustering_data import ClusteringData
+
+logger = logging.getLogger(__name__)
+
+
+class LabelNormalizationMethod(str, Enum):
+    """Supported label normalization methods."""
+    
+    SOFTMAX = "softmax"
+    L1 = "l1"
+    L2 = "l2"
+    SOFTMAX_TEMPERATURE = "softmax_temperature"
```

Comment:
These do the same thing! Remove SOFTMAX_TEMPERATURE since SoftmaxNormalizer already accepts temperature.